### PR TITLE
Community detection function

### PR DESCRIPTION
**Pull request by lassedochreden**
Hi @nilseling, 

here is the first version of the `detectCommunity` function. 

Happy to discuss!

### CODE REVIEW

File: R/detectCommunity.R [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+#' 
+#' @importFrom SingleCellExperiment colData
+#' @importFrom BiocParallel bplapply
+#' @importFrom igraph membership sizes
+#' @export
+
+
+detectCommunity <- function(object,
+                            colPairName,
+                            size_threshold = 10,
```
Comment (nilseling) [inline R/detectCommunity.R:80]: Should we enforce a threshold as default? Why mot keep it `0`?

**Review by nilseling — CHANGES_REQUESTED** [review]
Let's discuss these two points again. But well done! Love the docs!

**Comment by nilseling** [comment]
Great work! Nearly ready to merge. Do you mind bumping the version, adjusting NEWS and adding the function to the vignette?

File: R/detectCommunity.R [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                            BPPARAM = SerialParam()){
+  
+.valid.detectCommunity.input(object, colPairName, size_threshold, group_by, name, cluster_fun) 
+
+  if (!is.null(group_by)) {
+    cur_list <- bplapply(unique(colData(object)[,group_by]), function(x){
+      cur_object  <- object[,colData(object)[,group_by] == x]
+      cl_comm <- .detectCommunity_function(cur_object, colPairName, cluster_fun)
+      comm <- paste0(x,"_",membership(cl_comm))  
+      names(comm) <- colnames(cur_object)
```
[the remark this answers is not in the document]
Comment (lassedochreden) [reply R/detectCommunity.R:93]: Good point! For now, I added a **validity check**.

**Review by nilseling — APPROVED** [review]
Looks nice!

**Pull request merged**